torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,202 @@
+        return map;
+      }
+      map["input_size"] = at::IValue(input.sizes());
+      map["weight_size"] = at::IValue(weight.sizes());
+      map["stride"] = inputs[CONV2D_STRIDE];
+      map["padding"] = inputs[CONV2D_PADDING];
+      map["dilation"] = inputs[CONV2D_DILATION];
+      map["groups"] = inputs[CONV2D_GROUPS];
+    } else if (fname == "aten::mm") {
+      std::vector<int> tensors{0, 1};
+      bool check = validateInput(fname, 2, inputs, tensors);
+      if (!check) {
+        return map;
+      }
+
+      at::Tensor left = inputs[0].toTensor();
+      at::Tensor right = inputs[1].toTensor();
+      map["mat1_size"] = at::IValue(left.sizes());
+      map["mat2_size"] = at::IValue(right.sizes());
+    } else if (fname == "aten::mul.Tensor") {
+      std::vector<int> tensors{0};
+      bool check = validateInput(fname, 1, inputs, tensors);
+      if (!check) {
+        return map;
+      }
+
+      at::Tensor mat = inputs[0].toTensor();
+      map["mat_size"] = at::IValue(mat.sizes());
+    } else if (fname == "aten::add.Tensor") {
+      std::vector<int> tensors{0};
+      bool check = validateInput(fname, 1, inputs, tensors);
+      if (!check) {
+        return map;
+      }
+
+      at::Tensor mat = inputs[0].toTensor();
+      map["mat_size"] = at::IValue(mat.sizes());
+    }
+
+    return map;
+}
+
+uint64_t computeFlops(const std::string &op_name, const std::unordered_map<std::string, c10::IValue> &extra_args) {
+    if (op_name == "aten::conv2d") {
+      try {
+        auto input_sizes_ref = extra_args.at("input_size");
+        auto kernel_sizes_ref = extra_args.at("weight_size");
+        if (!input_sizes_ref.isIntList() || !kernel_sizes_ref.isIntList()) {
+          TORCH_WARN("Failed to compute flops for op aten::conv2d because it requires input and weight tensor sizes.");
+          return 0;
+        }
+
+        const std::vector<int64_t> input_sizes = input_sizes_ref.toIntVector();
+        const std::vector<int64_t> kernel_sizes = kernel_sizes_ref.toIntVector();
+        if (input_sizes.size() != 4 || kernel_sizes.size() != 4) {
+          TORCH_WARN("Failed to compute flops for op aten::conv2d because both input and weight must be size 4.");
+          return 0;
+        }
+        // format of the input is defined in torch.nn.quantized.functional.conv2d()
+        const uint64_t minibatch = input_sizes[0];
+        const uint64_t in_channels = input_sizes[1];
+        const uint64_t input_h = input_sizes[2];
+        const uint64_t input_w = input_sizes[3];
+        const uint64_t out_channels = kernel_sizes[0];
+        const uint64_t kernel_h = kernel_sizes[2];
+        const uint64_t kernel_w = kernel_sizes[3];
+        const uint64_t conv2d_multiply_factor = 2;
+
+        // grouping is NOT properly handled yet
+        return (conv2d_multiply_factor * minibatch * input_h * input_w * kernel_h * kernel_w * in_channels * out_channels);
```

Comment:
Nit, why do we need brackets here?
```suggestion
        return conv2d_multiply_factor * minibatch * input_h * input_w * kernel_h * kernel_w * in_channels * out_channels;
```